### PR TITLE
Bookdown draft

### PR DESCRIPTION
**Pull request by mikey-harper**
Hi @yihui ,

Here is my draft attempt at the book chapter. It is still slightly rough around the edges, but thought it would be best to get some feedback from you before going too far down the wrong route.

**Comments**
- Most the content is taken from the bookdown book but has required a lot of editing to squeeze it down. Generally, I have tried to cut out advanced concepts with the aim of providing details which are sufficient for users to get up and running.
- I have tried my hardest to match your writing style where I have written new content.
- I have added a short section on the typical project files: I think this a nice way of quickly explaining how all the parts link, and is something I remember wanting when I started learning bookdown.
- I generally have provided cross-references to the bookdown book where more detail is provided about advanced concepts. I have tried not to do this too often as it is important that this is lives up to the name and is a definitive guide for most users though.
- I have started preparing cross-references to other parts of the book which are currently missing (Figures, customization)
- I have still yet to write the `Some tips` section, but I think it would be good to end the chapter with some practical tips on usage.

**General questions:**

- [ ]  Can you let me know any improvements I can make to emulate your writing style?
- [ ] I still feel it is slightly too long, so would like to know your thoughts on the overall length.
- [ ] Do you think there are any key topics which I should not have removed?

**Next steps**:
- Shorten the content from `Editing` onwards: I didn't have time to fully cut this down from the book.

Thanks in advance for any help!

### CODE REVIEW

File: 12-bookdown.Rmd [hunk 1]
```diff
@@ -1,3 +1,630 @@
...
+# or GitHub
+devtools::install_github('rstudio/bookdown')
+```
+
+If you use RStudio, you can start a new bookdown project from the menu `File -> New Project... -> New Directory -> Book Project using bookdown`.^[Alternatively, the command `bookdown:::bookdown_skeleton(getwd())` (N.B. `:::`) will create a skeleton project in your current working directory.] Open the R Markdown file `index.Rmd` and click the button Build Book on the Build tab of RStudio. Now you should see the index page of this book demo in the RStudio Viewer, which looks like Figure \@ref(fig:bookdown-template).  
+
+You may add or change the R Markdown files, and hit the `Knit` button again to preview the book. If you prefer not to use RStudio, you may also compile the book through the command line. See Section \@ref(usage) for details.
+
+```{r bookdown-template, echo=FALSE, fig.cap= "The HTML output of the bookdown template", out.width='100%'}
+knitr::include_graphics("images/bookdown-template.png")
```
Comment (yihui) [inline 12-bookdown.Rmd]: Please add the argument `dpi = NA`.

If you have not optimized the image, I often use http://optimizilla.com to minimize the file size of images.

File: 12-bookdown.Rmd [hunk 2]
```diff
@@ -1,3 +1,630 @@
...
+```
+
+## Project Structure
```
Comment (yihui) [inline 12-bookdown.Rmd]: S -> s

File: 12-bookdown.Rmd [hunk 3]
```diff
@@ -1,3 +1,630 @@
...
+├── _output.yml
+├──  book.bib
+├──  index.Rmd
+├──  preamble.tex
+├──  README.md
+└──  style.css
+```
+<!---
+Potential to replace diagram by extracting file structure directly from the package 
+list.files(bookdown:::bookdown_file("rstudio", "templates", "project", "resources"))
```
Comment (yihui) [inline 12-bookdown.Rmd]: The hard-coded list is fine. I'd put `index.Rmd` in the beginning.

File: 12-bookdown.Rmd [hunk 4]
```diff
@@ -1,3 +1,630 @@
...
+--->
+
+As a summary of these files:
+
+- `index.Rmd`: This is the only Rmd document to contain a YAML frontmatter as described within the \@ref(basics), and is the first book chapter.
+- `Rmd files`: a typical **bookdown** book contains multiple chapters, and one chapter lives in one Rmd file.
```
Comment (yihui) [inline 12-bookdown.Rmd]: No backticks on `Rmd files`.

File: 12-bookdown.Rmd [hunk 5]
```diff
@@ -1,3 +1,630 @@
...
+- `_bookdown.yml`: settings to control how bookdown builds the book.
```
Comment (yihui) [inline 12-bookdown.Rmd]: bold package names: `**bookdown**`

File: 12-bookdown.Rmd [hunk 6]
```diff
@@ -1,3 +1,630 @@
...
+- `output.yml`: specifies the formatting of the HTML, LaTeX/PDF and e-books.
```
Comment (yihui) [inline 12-bookdown.Rmd]: missing underscore: `_output.yml`

File: 12-bookdown.Rmd [hunk 7]
```diff
@@ -1,3 +1,630 @@
...
+- `preamble.tex` & `style.css`: allows for the appearance and styles of the files to be adjusted. Adjusting these are beyond the scope of this chapter, but are covered in Section \@ref(customization).
```
Comment (yihui) [inline 12-bookdown.Rmd]: `&` -> `and`

File: 12-bookdown.Rmd [hunk 8]
```diff
@@ -1,3 +1,630 @@
...
+
+<!--- Add a chapter on customization --->
+
+These files are explained in greater detail within the following subsections. 
+
+### index File
```
Comment (yihui) [inline 12-bookdown.Rmd]: i -> I; F -> f

File: 12-bookdown.Rmd [hunk 9]
```diff
@@ -1,3 +1,630 @@
...
+rmd_files: ["index.Rmd", "abstract.Rmd", "intro.Rmd"]
+```
+
+In this case, **bookdown** will just use whatever you defined in this YAML field without any special treatments of `index.Rmd` or underscores. If you want both HTML and LaTeX/PDF output from the book, and use different Rmd files for HTML and LaTeX output, you may specify these files for the two output formats separately, e.g.,
+
+```yaml
+rmd_files:
+  html: ["index.Rmd", "abstract.Rmd", "intro.Rmd"]
+  latex: ["abstract.Rmd", "intro.Rmd"]
+```
```
Comment (yihui) [inline 12-bookdown.Rmd]: I'd rather leave this out. Too deep detail.

File: 12-bookdown.Rmd [hunk 10]
```diff
@@ -1,3 +1,630 @@
...
+      position: static
+bookdown::pdf_book:
+  keep_tex: yes
+bookdown::html_book:
+  css: toc.css
+```
+
+Section \@ref(bookdown-output) provides further detail on how the output formats can be customised.
+
+## R Markdown Extensions
```
Comment (yihui) [inline 12-bookdown.Rmd]: `R Markdown Extensions` -> `Markdown extensions`

**Comment by mikey-harper** [comment]
Okay, thank you for this feedback. I did feel I had gone a bit off tangent, hence my request for some input.

Sorry for copying a lot. I guess trying to write like you made me nervous to generate too much new content, but went too far the other way!

As you warned, the process can be a bit difficult to start, but I hope I can soon feel less of a hindrance and more of a help to writing this book. 

Will come back shortly with an amended version.

**Comment by yihui** [comment]
No worries. Thanks!

**Comment by mikey-harper** [comment]
First set of revisions made: have rewritten most of the content. In doing so I realised how bad some parts of it was!

Not quite ready for a review : I will go through things again tomorrow evening as there are some bits I wanted to look at with some fresh eyes tomorrow. Will let you know when I am ready for your comments.

Thanks for the feedback. and suggestions to writing to a limited time schedule. This certainly is a good way of think about things! Learning to write at the level of efficiency you do is certainly an unrealistic aim of mine.

**Comment by mikey-harper** [comment]
I hope this is looking more like you had in mind :) I have rewritten most the content from scratch.

P.S. I am not sure if I am using GitHub 100% correctly as this is my first proper pull request.

**Comment by yihui** [comment]
Looks much better!! I'll review it after I finish the chapter I'm currently working on (hopefully tomorrow morning).

**Comment by mikey-harper** [comment]
Fantastic, thank you. 

Are you working on the blogdown Chapter? I am keen to look at another chapter which should be quicker this time around.Will draft an outline of one of the remaining chapters.

**Comment by yihui** [comment]
Yes, I'm working on blogdown.

Please feel free to go ahead with another chapter.

**Comment by yihui** [comment]
Instead of giving a line-by-line review, I think it will be much faster if I just edit your draft directly. If you are okay with that, I'll push to your branch to update this PR.

**Comment by mikey-harper** [comment]
Yes, happy with that. It sounds easier for both of us.

**Pull request merged**